### DISCUSSION TITLE
constrain listbox dragging and selection to handles/triggers within listboxitem?

### DISCUSSION THREAD

**Discussion by binaryartifex** [adobe/react-spectrum#6572] [Q&A] [answered]
as can be seen from the image below, i have a listbox of column rows that can be dragged to change column order and selected to show/hide it in the table. is there a way to limit the enabling of the draggable feature to only when the grab button with the dots is pressed and only selected when the checkbox is ticked? or does me having this things on each arrow essentially just assist in indicating what can be done? 

![image](https://github.com/adobe/react-spectrum/assets/68464707/07a617c1-bb73-4f87-9b7c-27665fc2387d)

**Comment by snowystinger** [comment, marked as answer]
If you wanted to separate them out, you could use GridList instead. Then depending on the combination of actions/dragging/selection they will act more independently from each other https://react-spectrum.adobe.com/react-aria/GridList.html#row-actions

The reason why you can't separate them in ListBox is explained here https://react-spectrum.adobe.com/react-aria/ListBox.html#text-slots so in a ListBox, they are as you've noted, an indication of what can be done.

**Reply by binaryartifex** [reply]
right you are. was staring me right in the face. cheers for taking the time mate.
